sycl/include/sycl/exception.hpp

```diff
@@ -66,6 +67,9 @@ class __SYCL_EXPORT SYCLCategory : public std::error_category {
   const char *name() const noexcept override { return "sycl"; }
   std::string message(int) const override { return "SYCL Error"; }
 };
+
+// Forward declare to declare as a friend in sycl::excepton.
+__SYCL_EXPORT pi_int32 get_pi_error(const exception &e);
```

Comment:
I'm not sure if it should be an exported symbol or inline header-only utility. Or, in other words (I think), if layout of `sycl::exception` is a part of ABI.

Reply:
We do have https://github.com/intel/llvm/blob/sycl/sycl/test/abi/layout_exception.cpp, but then it's unclear to me why `get_cl_code` wasn't an `inline` function...